src/Veldrid/BlendAttachmentDescription.cs

```diff
@@ -12,6 +12,10 @@ public struct BlendAttachmentDescription : IEquatable<BlendAttachmentDescription
         /// </summary>
         public bool BlendEnabled;
         /// <summary>
+        /// Controls which components of the color will be written to the framebuffer.
+        /// </summary>
+        public ColorWriteMask ColorWriteMask;
```

Comment:
The only concern I have is that if you have a default-constructed BlendAttachmentDescription, then this is a breaking behavioral change, and a big one at that. E.g. if I am already doing this:

```
BlendAttachmentDescription desc = default();
desc.... = ...;
// etc.
UseDesc(desc);
```

I will now be using an empty ColorWriteMask and nothing will work anymore.

It's ugly, but in the name of backwards compatibility we need to make this a nullable field (`ColorWriteMask?`) so that default-constructed objects still work correctly. Nothing else about the constructors you modified need to be adjusted (they can still assign a non-null value to the field, as long as the behavior is the same as before).

Then, I'd introduce a helper function that converts from a `ColorWriteMask?` to `ColorWriteMask`, understanding that a `null` mask is the same as a mask with `.All` assigned, and use that in all the other places that are reading from the BlendAttachmentDescription field. Everything else would stay the same.

Reply:
Good note, have made it nullable as requested. I'm not familiar with the helper patterns in Veldrid, but I've added a public extension method (`GetOrDefault`) for it. Should it be internal instead? or a non-extension static method?